src/components/CreateJobPostForm/CreateJobPostForm.tsx

```diff
@@ -0,0 +1,208 @@
+        : createDataForResolver(data, createJobPostValidationSchema);
+
+      return {
+        values: validData,
+        errors: errorsMessages,
+      };
+    },
+  });
+  const fileTypes: string[] = Object.values(createJobAttachmentFileTypes);
+  const MAX_FILE_SIZE = 5e6;
+
+  useEffect(() => {
+    if (!postDraftLoading) {
+      if (postDraftSuccess) toast.success(t("PostJob.messages.draftSuccess"));
+      if (postDraftError) toast.error(t("PostJob.messages.draftError"));
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [postDraftSuccess, postDraftError]);
+
+  useEffect(() => {
+    if (!postJobLoading) {
+      if (postJobSuccess) toast.success(t("PostJob.messages.postSuccess"));
+      if (postJobError) toast.error(t("PostJob.messages.postError"));
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [postJobSuccess, postJobError]);
+
+  useEffect(() => {
+    setValue("file", selectedFile);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedFile]);
+
+  const handlerSubmitForm = (data: IJobPostFormFields) => {
+    const { skills: skillsId, jobDescription, rate, duration, ...rest } = data;
+
+    const objToRequest = {
+      ...rest,
+      skillsId: skillsId.length >= 3 ? skillsId : null,
+      jobDescription: jobDescription || null,
+      rate: rate || null,
+      duration: duration || undefined,
+      userId: userData?.id!,
+      isDraft,
+    };
+
+    if (isDraft) {
+      postDraft(objToRequest);
+      return;
+    }
+
+    const formData = new FormData();
+
+    // eslint-disable-next-line array-callback-return
+    Object.entries(objToRequest).map(([key, value]) => {
+      if (Array.isArray(value)) {
+        value.map((element, index) =>
+          formData.append(`${key}[${index}]`, `${element}`)
+        );
+      } else {
+        formData.append(key, value instanceof File ? value : `${value}`);
+      }
+    });
+
+    postJobPost(formData);
+  };
+
+  const handlerPostDraft = () => setIsDraft(() => true);
+  const handlerPostJobPost = () => setIsDraft(() => false);
+
+  const handleUploadFile: ChangeEventHandler<HTMLInputElement> = ({
+    target,
+  }) => {
+    if (target.files && target.files[0]) {
+      const file = target.files[0];
+
+      if (fileTypes.includes(file.type) && file.size < MAX_FILE_SIZE) {
+        setSelectedFile(file);
+      } else if (!fileTypes.includes(file.type)) {
+        toast.error(t("PostJob.messages.unsupportedFile"));
+      } else {
+        toast.error(t("PostJob.messages.bigFileSize"));
+      }
+    }
+  };
+
+  return (
+    <SWrapper>
+      <h2>Create a job post</h2>
+      <form onSubmit={handleSubmit(handlerSubmitForm)}>
+        {propsDataCollection.map((propsData) => (
+          <LayoutElementWithTitle
+            key={propsData.title}
+            control={control as unknown as Control}
+            errors={validationErrors as unknown as FieldErrors}
```

Comment:
validationErrors as unknown as FieldErrors why its so? it not look like right code)

Reply:
It's errors from hook form, just renamed

![Screenshot from 2022-09-09 18-25-06](https://user-images.githubusercontent.com/92886035/189386554-1dc7f84a-e163-43b3-93af-17f473d32699.png)

Reply:
ask PM to create a tech task for refactoring when anyone will have time it's better to do it, I would prefer not left as it is